pkg/cmd/server/server.go

```diff
@@ -440,6 +443,50 @@ func (s *server) run() error {
 	return nil
 }
 
+// setupBeforeControllerRun do any setup that needs to happen before the controllers are started.
+func (s *server) setupBeforeControllerRun() error {
+	client, err := ctrlclient.New(s.mgr.GetConfig(), ctrlclient.Options{Scheme: s.mgr.GetScheme()})
+	// the function is called before starting the controller manager, the embedded client isn't ready to use, so create a new one here
+	if err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to create client")
```

Comment:
The error will be logged by the upper caller, no need to log it here

Reply:
removed